components/camera/client.go

```diff
@@ -154,6 +150,10 @@ func (c *client) Stream(
 			select {
 			case <-streamCtx.Done():
 				return
+			case <-c.stopStreamsCh:
+				if err := stream.Close(ctxWithMIME); err != nil {
+					panic(err)
+				}
```

Comment:
seems like we can just return here like the case for `streamCtx.Done()`? also don't think we should be panicking regardless

Reply:
that's fair - I threw in the panic because `Close` just cancels an internal context and therefore should always return `nil`. if this ever changes that would unexpected and maybe we'd want to know about that.

that said I'm happy to just return

Reply:
let's just return so behavior is consistent, and seems like that's been working fine

Reply:
updated to just be warning log: https://github.com/viamrobotics/rdk/pull/3467/commits/d1727b570847882cf1c1f36e89bce43969810eb7